lib/broccoli/ember-app.js

```diff
@@ -669,59 +697,112 @@ EmberApp.prototype._filterAppTree = function() {
     return this._cachedFilterAppTree;
   }
 
-  var podPatterns = this._podTemplatePatterns();
-  var excludePatterns = podPatterns.concat([
-    // note: do not use path.sep here Funnel uses
-    // walk-sync which always joins with `/` (not path.sep)
+  var appExcludePatterns = [
     'styles/**/*',
     'templates/**/*',
-  ]);
-
-  this._cachedFilterAppTree = new Funnel(this.trees.app, {
-    exclude: excludePatterns,
-    annotation: 'Funnel: Filtered App'
-  });
+  ];
+  this._cachedFilterAppTree = this._filterAnyAppTree(this.trees.app, appExcludePatterns, 'App');
 
   return this._cachedFilterAppTree;
 };
 
-EmberApp.prototype.podTemplates = function() {
-  return new Funnel(this.trees.app, {
+/**
+  Filters templates from the `fastboot/app` tree.
+
+  @private
+  @method _filterFastbootAppTree
+  @return {Tree}
+*/
+EmberApp.prototype._filterFastbootAppTree = function() {
+  if (this._cachedFilterFastbootAppTree) {
+    return this._cachedFilterFastbootAppTree;
+  }
+
+  if (this.trees.fastbootApp) {
+    var appExcludePatterns = [
+      'templates/**/*',
+    ];
+    this._cachedFilterFastbootAppTree = this._filterAnyAppTree(this.trees.fastbootApp, appExcludePatterns, 'Fastboot App');
+
+    return this._cachedFilterFastbootAppTree;
+  }
+};
+
+EmberApp.prototype.podTemplates = function(tree, destDir) {
+  return new Funnel(tree, {
     include: this._podTemplatePatterns(),
     exclude: [ 'templates/**/*' ],
-    destDir: this.name + '/',
+    destDir: destDir,
     annotation: 'Funnel: Pod Templates'
   });
 };
 
-EmberApp.prototype._templatesTree = function() {
-  if (this._cachedTemplateTree) {
-    return this._cachedTemplateTree;
-  }
+/**
+  Filters given templates tree into the app tree.
 
+  @private
+  @param {String} destDirPath the prefix path where templates are written
+  @param {Tree} templatesTree
+  @param {Tree} appTree
+  @method _templatesAnyTree
+ */
+EmberApp.prototype._templatesAnyTree = function(destDirPath, templatesTree, appTree) {
```

Comment:
Does fastboot currently allow/enable shipping alternative templates for fastboot? My understanding, was this feature was only to add new JS  to slightly augment or replace modules. Such as `ajax` -> `njax`